### PR TITLE
💻 Adding styled-components and initial design system

### LINKED ISSUE #55: Create frontend design system
Create design system using styled-components

### LINKED ISSUE #65: Design System
Front-end: Implement design system using styled-components

### PR DESCRIPTION
**Pull request by lesliexin**
## Purpose
Closes #65. Adding the `styled-components` library and an initial design system.

## Approach
Creating the following files to store constants used throughout the entire application:
- `device.ts` -> various device sizes for responsivity
- `typography.ts` -> design system types
- `colours.ts` -> design system colours

#### Blog Posts
- [Comparing frontend styling options](https://www.notion.so/Frontend-Styling-9546cad80aee43619ca85c1344e6d870)

### CODE REVIEW

File: client/styles/colours.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+// Primary colors
```
Comment (xujustinj) [inline client/styles/colours.ts:1]: If we use `colours` as the name of the object, then we should go all out 🚀 

```suggestion
// Primary colours
```

File: client/styles/colours.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+const purple = "#47209D";
+const blueDark = "#3370EC";
+const greenDark = "#3C8A44";
+
+// Secondary colors
```
Comment (xujustinj) [inline client/styles/colours.ts:6]:
```suggestion
// Secondary colours
```

File: client/styles/colours.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+const blueLight = "#D8F7FE";
+const purpleLight = "#E5D0FF";
+const redLight = "#F9CED7"
+const yellowLight = "#FEF5D8";
+const greenLight = "#D8FEF7";
+
+// Neutral colors
```
Comment (xujustinj) [inline client/styles/colours.ts:13]:
```suggestion
// Neutral colours
```

File: client/styles/device.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+const size = Object.freeze({
```
Comment (xujustinj) [inline client/styles/device.ts:1]: if this is specifically used for screen widths and not heights, maybe `width` is a better name?

**Review by xujustinj — CHANGES_REQUESTED** [review]
`client/package-lock.json` shouldn't exist; try deleting the file and running `npm i` from the root folder of the repo (outside of `client`)

File: client/styles/colours.ts (hunk 1 above)
Comment (lesliexin) [reply client/styles/colours.ts:1]: oops good catch!

File: client/styles/device.ts (hunk 4 above)
Comment (lesliexin) [reply client/styles/device.ts:1]: ya that makes sense :)

Comment (lesliexin) [reply client/styles/device.ts:1]: actually, I'm looking at it again and I feel like `size` makes more sense as we could in the future export heights if need be from the same set of constants

Comment (jeffzh4ng) [reply client/styles/device.ts:1]: Would these sizes be reusable for heights though? Or should we have two objects, `width` and `height`?

**Review by jeffzh4ng — CHANGES_REQUESTED** [review]

File: client/styles/device.ts (hunk 4 above)
Comment (lesliexin) [reply client/styles/device.ts:1]: hmm ya true

**Review by jeffzh4ng — APPROVED** [review]

File: client/package.json [hunk 5]
```diff
@@ -12,12 +12,14 @@
...
   "dependencies": {
+    "@types/styled-components": "^5.1.7",
     "babel-jest": "^26.6.3",
     "fuse.js": "^6.4.6",
     "next": "10.0.3",
     "react": "17.0.1",
     "react-dom": "17.0.1",
-    "vercel-is-pull-request": "^0.0.8"
+    "vercel-is-pull-request": "^0.0.8",
+    "styled-components": "^5.2.1"
```
Comment (xujustinj) [inline client/package.json:22]: Let's keep alphabetical order

```suggestion
    "styled-components": "^5.2.1",
    "vercel-is-pull-request": "^0.0.8"
```

**Review by ShreyasPrasad — DISMISSED** [review]
lgtm

**Review by xujustinj — APPROVED** [review]

**Review by jeffzh4ng — APPROVED** [review]

**Pull request merged**